client/jetpack-cloud/sections/sidebar-navigation/lib/sidebar.ts

```diff
@@ -1,14 +1,30 @@
 import page from 'page';
 import { itemLinkMatches } from 'calypso/my-sites/sidebar/utils';
+import {
+	JETPACK_MANAGE_ASSIGN_LICENCE_LINK,
+	JETPACK_MANAGE_ISSUE_LICENCE_LINK,
+	JETPACK_MANAGE_LICENCES_LINK,
+} from './constants';
 
 /**
  * Determines whether a menu item should be selected based on the current URL path.
  * @param link The link to compare against the current URL path.
+ * @param path The current URL path.
  * @returns A boolean value indicating whether the link matches the current URL path.
  */
-export const isMenuItemSelected = ( link: string ) => {
-	const pathname = window.location.pathname;
-	return itemLinkMatches( link, pathname );
+export const isMenuItemSelected = ( link: string, path: string ) => {
+	let overridenPath = path;
+
+	// For Assign and Issue license path, we will override it to be license path to make the menu selected.
+	if (
+		( link === JETPACK_MANAGE_LICENCES_LINK &&
+			path.startsWith( JETPACK_MANAGE_ASSIGN_LICENCE_LINK ) ) ||
+		path.startsWith( JETPACK_MANAGE_ISSUE_LICENCE_LINK )
```

Comment:
This now means the overridenPath will be set to `JETPACK_MANAGE_LICENCES_LINK` only 
 
- When link is equal to JETPACK_MANAGE_LICENCES_LINK and path starts with JETPACK_MANAGE_ASSIGN_LICENCE_LINK.
- When the path starts with JETPACK_MANAGE_ISSUE_LICENCE_LINK.

Does this have to be written as:

```suggestion
		link === JETPACK_MANAGE_LICENCES_LINK &&
		( path.startsWith( JETPACK_MANAGE_ASSIGN_LICENCE_LINK ) ||
			path.startsWith( JETPACK_MANAGE_ISSUE_LICENCE_LINK ) )
```